examples/split_plot_rcbd.R

```diff
@@ -0,0 +1,122 @@
+# This is an example of a split-plot RCBD experimental design based off of the first
+# example in [1] and further analysis in [2]. The data can be downloaded from
+# [3].
+#
+# To run the script type:
+#
+# source('split_plot_rcbd.R', print.eval = TRUE)
+#
+# [1] http://www.unh.edu/halelab/BIOL933/labs/lab9.pdf
+# [2] http://www.personal.psu.edu/mar36/stat_461/split_plot/split_plot.html
+# [3] http://www.unh.edu/halelab/BIOL933/schedule.htm
+#
+# TODO : Maybe I should use the `data('plots')` from the agricolae package
+# instead?
+#
+# NOTE : The code that will be shown to the user in the app is bounded by these
+# separators: #------------#.
+
+# Load the necessary libraries.
+#-----------------------------------------------------------------------------#
+library('agricolae')
+library('effects')  # for allEffects()
+#-----------------------------------------------------------------------------#
+
+sep <- function(n){
+  # This function simply prints a line of "=" to the screen as a separator and
+  # will not show up in the app. It is only here for nice console printing.
+  line <- paste0(paste(replicate(n, "="), collapse = ""), '\n')
+  cat(line)
+}
+
+# Load the data.
+# repetitions: block
+# main plots: SeedLotA (four plots in a block)
+# subplots: TrtmtB (four subplots in a main plot)
+#-----------------------------------------------------------------------------#
+my.data <- read.csv('Lab9ex1.csv')
+#-----------------------------------------------------------------------------#
+
+# The factors in this experiment are recorded with numerical values so they need
+# to be recoded as factors. The user will have to set this manually in the app.
+# NOTE : This is not necessary if `aov()` is used instead of `lm()`.
+#-----------------------------------------------------------------------------#
+my.data$SeedLotA <- as.factor(my.data$SeedLotA)
+my.data$Block <- as.factor(my.data$Block)
+my.data$TrtmtB <- as.factor(my.data$TrtmtB)
+#-----------------------------------------------------------------------------#
+
+# This is the standard model for a split plot RCBD.
+# NOTE : I'm getting this warning message:
```

Comment:
I am looking into this error message. Despite the error, I still get the same summary output as they did. There's a lot of questions online about this, and my hunch is that it's OK that it's singular. I'm not sure if it's this particular dataset or if I'd get this with any dataset with this experimental design and degrees of freedom.

I emailed the author (Iago) of the materials and will hopefully hear back soon.